lib/internal/Magento/Framework/File/Uploader.php

```diff
@@ -803,7 +803,7 @@ public static function getNewFileName($destinationFile)
         $fileInfo = pathinfo($destinationFile);
         $index = 1;
         while ($fileExists($fileInfo['dirname'] . '/' . $fileInfo['basename'])) {
-            $fileInfo['basename'] = $fileInfo['filename'] . '_' . $index++ . '.' . $fileInfo['extension'];
+            $fileInfo['basename'] = (isset($fileInfo['extension'])) ? $fileInfo['filename'] . '_' . $index++ . '.' . $fileInfo['extension'] : $fileInfo['filename'] . '_' . $index++;
```

Comment:
```suggestion
            $fileInfo['basename'] = $fileInfo['filename'] . '_' . $index++ . (isset($fileInfo['extension']) ? '.' . $fileInfo['extension'] : '');
```

Reply:
I believe that suggested update look little bit clear.
@matthieu-barbe what do you think?

Reply:
It's perfect ;)